StRoot/StPicoDstMaker/StPicoDstMaker.cxx

```diff
@@ -854,10 +854,11 @@ Int_t StPicoDstMaker::MakeWrite() {
 
   mBField = muEvent->magneticField();
 
+#if !defined (__TFG__VERSION__)
   // Get Emc collection
   mEmcCollection = mMuDst->emcCollection();
-  StMuEmcUtil *emcUtil = new StMuEmcUtil();
   if ( !mEmcCollection ) {
+    static StMuEmcUtil *emcUtil = new StMuEmcUtil();
     // Recover StEmcCollection in case of broken/deleted pointer
     // This usually happens during daq->picoDst converstion
     mEmcCollection = emcUtil->getEmc( mMuDst->muEmcCollection() );
```

Comment:
Hm... I don't see how you deal with the objects allocated by `StMuEmcUtil::getEmc()` on every call when the argument is a nullptr. Isn't it what was causing the memory leak and something we discussed somewhere else?

Defining `mEmcCollection` as `std::unique_ptr<StEmcCollection> mEmcCollection` is probably  the way it should be done in the modern C++

Reply:
> Hm... I don't see how you deal with the objects allocated by StMuEmcUtil::getEmc() on every call when the argument is a >nullptr. Isn't it what was causing the memory leak and something we discussed somewhere else?
I tested both ways and those do not produce memory leak (at least I do not see it). I removed the pointer in the last commit. Should be okay now. 

As for the modern C++ style. I generally agree with it but this may need more investigation of the instance ownership. For example, the StMuEmcUtil class allocates the memory with the new operator but does not free it in the destructor, assuming (I guess) that the codes that call getEmc(...) will own the instance. Freeing the memory with the delete command causes segfault because of some StRoot issues.

Anyhow, I would ask you or Gene to run the current version of the code and see whether it causes the memory leak or not.